salt/modules/vsphere.py

```diff
@@ -3519,3 +3532,140 @@ def _set_syslog_config_helper(host, username, password, syslog_config, config_va
         ret_dict.update({'syslog_restart': {'success': response['retcode'] == 0}})
 
     return ret_dict
+
+
+def add_host_to_vsphere(host, username, password, vsphere_protocol=None,
+                        vsphere_port=None, host_names=None, host_username=None,
+                        host_password=None):
+    '''
+    Add an ESX host to vSphere.
+    :param host: vSphere hostname or IP
+    :param username: Login to vSphere with this username
+    :param password: Login to vSphere with this password
+    :param vsphere_protocol:
+    :param vsphere_port:
+    :param host_names:
+    :param host_username:
+    :param host_password:
+    :return:
```

Comment:
Return doesn't have anything provided here.

Also, a CLI example would be super!